master/internal/core.go

```diff
@@ -375,14 +374,14 @@ func (m *Master) Run() error {
 		provisionerSlotsPerInstance = p.SlotsPerInstance()
 	}
 
-	proxyRef, _ := m.system.ActorOf(actor.Addr("proxy"), &proxy.Proxy{})
+	m.proxy, _ = m.system.ActorOf(actor.Addr("proxy"), &proxy.Proxy{})
 
 	cert, err := m.readTLSCertificate()
 	if err != nil {
 		return errors.Wrap(err, "failed to read TLS certificate")
 	}
 
-	m.initializeResourceProviders(proxyRef, provisionerSlotsPerInstance, cert)
+	m.initializeResourceProviders(m.proxy, provisionerSlotsPerInstance, cert)
```

Comment:
question: do we still need to pass the proxy in to the RPs?

Reply:
No. will remove